src/parser.rs

```diff
@@ -1095,6 +1094,9 @@ fn parse_string_size_expression(lexer: &mut ParseSession) -> Option<AstNode> {
             let size_expr = parse_expression(lexer);
             let error_range = lexer.source_range_factory.create_range(opening_location..lexer.range().end);
 
+            // Don't emit warnings if this looks like an enum (will be caught by validation)
+            let is_enum_like = matches!(size_expr.get_stmt(), AstStatement::ExpressionList(_));
```

Comment:
Can you give some more context here, e.g. needed because of `TYPE InvalidEnum2 : STRING (a := 1, b := 2); END_TYPE`